antsibull/cli/doc_commands/sphinx_init.py

```diff
@@ -0,0 +1,95 @@
+# Author: Felix Fontein <felix@fontein.de>
+# License: GPLv3+
+# Copyright: Ansible Project, 2021
+"""Entrypoint to the antsibull-docs script."""
+
+import os
+import os.path
+
+from ... import app_context
+from ...jinja2.environment import doc_environment
+from ...logging import log
+
+
+mlog = log.fields(mod=__name__)
+
+
+TEMPLATES = [
+    '.gitignore',
+    'build.sh',
+    'conf.py',
+    'requirements.txt',
+    'rst/index.rst',
+]
+
+
+def write_file(filename: str, content: str) -> None:
+    """
+    Write content into a file.
+    """
+    if os.path.exists(filename):
+        with open(filename, 'r') as f:
+            existing_content = f.read()
+        if existing_content == content:
+            print(f'Skipping {filename}')
+            return
+
+    print(f'Writing {filename}...')
+    with open(filename, 'w') as f:
+        f.write(content)
+
+
+def site_init() -> int:
+    """
+    Initialize a Sphinx site template for a collection docsite.
+
+    Creates a Sphinx configuration file, requirements.txt and a bash script which uses
+    antsibull-docs to build the RST files for the specified collections.
+
+    :returns: A return code for the program.  See :func:`antsibull.cli.antsibull_docs.main` for
+        details on what each code means.
+    """
+    flog = mlog.fields(func='site_init')
+    flog.notice('Begin site init')
+
+    app_ctx = app_context.app_ctx.get()
+
+    dest_dir = app_ctx.extra['dest_dir']
+    collections = app_ctx.extra['collections']
+    collection_version = app_ctx.extra['collection_version']
+    use_current = app_ctx.extra['use_current']
+    squash_hierarchy = app_ctx.extra['squash_hierarchy']
+    lenient = app_ctx.extra['lenient']
+
+    if os.path.exists(dest_dir) and not os.path.isdir(dest_dir):
+        print(f'Expecting {dest_dir} to be a directory')
+        return 3
+    os.makedirs(dest_dir, exist_ok=True)
```

Comment:
It looks like these makedirs calls with exist_ok=True will be insecure since the checks that dest_dir is only writable by the user are being skipped.

Reply:
`dest_dir` cannot be writable by the user if it does not exist. Right now the code does not make assumptions on whether `dest_dir` exists (as opposed the other antsibull-docs subcommands).

Reply:
c361cda84d9a533af973909e40f80b6403b84e8d removed this code and requires `dest_dir` to already exist.